### PR TITLE
Updated Documentation for the Review endpoints

### PR DESCRIPTION
**Pull request by iursevla**
## Proposed change

This PR improves the documentation of the `Review` controller endpoints

<details>
<summary>Examples</summary>

image

image

</details>

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [X] Code quality improvements to existing code

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 

## Checklist

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

File: frigate/api/defs/severity_enum.py
Comment (NickM-27) [inline frigate/api/defs/severity_enum.py:1]: this already exists, we should not be recereating it https://github.com/blakeblackshear/frigate/blob/edaccd86d61d70baa34266d47faf7c8aa0eed50e/frigate/review/maintainer.py#L42

File: frigate/api/review.py
Comment (NickM-27) [inline frigate/api/review.py:559]: why is this being changed? it should stay the same just like motion

**Review by NickM-27 — CHANGES_REQUESTED** [review]
I think we should go ahead and create sub folders in defs for `response` and `request` to keep them more separate

**Comment by NickM-27** [comment]
sorry, didn't realize it was still in draft

File: frigate/api/defs/severity_enum.py
Comment (iursevla) [reply frigate/api/defs/severity_enum.py:1]: But that enum does not contain`significant_motion`. Is it okay to add it to the enum you've linked?

```py
class SeverityEnum(str, Enum):
    alert = "alert"
    detection = "detection"
    significant_motion = "significant_motion"
```

**Comment by iursevla** [comment]
> sorry, didn't realize it was still in draft, the email link took me right to the file changes

No worries. Those were good comments

File: frigate/api/defs/severity_enum.py
Comment (NickM-27) [reply frigate/api/defs/severity_enum.py:1]: significant_motion does not exist currently so there is no reason for it to be added

File: frigate/api/review.py
Comment (iursevla) [reply frigate/api/review.py:559]: I think it's due to the 'nan' values:

image


```text
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/uvicorn/protocols/http/h11_impl.py", line 406, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/usr/local/lib/python3.9/dist-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/applications.py", line 113, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 189, in __call__
    response_sent.set()
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.9/dist-packages/starlette/_utils.py", line 83, in collapse_excgroups
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/usr/local/lib/python3.9/dist-packages/slowapi/middleware.py", line 136, in dispatch
    response = await call_next(request)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 189, in __call__
    response_sent.set()
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.9/dist-packages/starlette/_utils.py", line 83, in collapse_excgroups
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/workspace/frigate/frigate/api/fastapi_app.py", line 78, in frigate_middleware
    response = await call_next(request)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 189, in __call__
    response_sent.set()
  File "/usr/lib/python3.9/contextlib.py", line 135, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.9/dist-packages/starlette/_utils.py", line 83, in collapse_excgroups
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 187, in __call__
    response = await self.dispatch_func(request, call_next)
  File "/usr/local/lib/python3.9/dist-packages/starlette_context/middleware/context_middleware.py", line 78, in dispatch
    response = await call_next(request)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 163, in call_next
    raise app_exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/base.py", line 149, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/usr/local/lib/python3.9/dist-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/usr/local/lib/python3.9/dist-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/usr/local/lib/python3.9/dist-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/usr/local/lib/python3.9/dist-packages/starlette/routing.py", line 73, in app
    response = await f(request)
  File "/usr/local/lib/python3.9/dist-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
  File "/usr/local/lib/python3.9/dist-packages/fastapi/routing.py", line 214, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/usr/local/lib/python3.9/dist-packages/starlette/concurrency.py", line 39, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/usr/local/lib/python3.9/dist-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
  File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 2405, in run_sync_in_worker_thread
    return await future
  File "/usr/local/lib/python3.9/dist-packages/anyio/_backends/_asyncio.py", line 914, in run
    result = context.run(func, *args)
  File "/workspace/frigate/frigate/api/review.py", line 560, in audio_activity
    return JSONResponse(content=normalized)
  File "/usr/local/lib/python3.9/dist-packages/starlette/responses.py", line 178, in __init__
    super().__init__(content, status_code, headers, media_type, background)
  File "/usr/local/lib/python3.9/dist-packages/starlette/responses.py", line 41, in __init__
    self.body = self.render(content)
  File "/usr/local/lib/python3.9/dist-packages/starlette/responses.py", line 181, in render
    return json.dumps(
  File "/usr/lib/python3.9/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/lib/python3.9/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.9/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
ValueError: Out of range float values are not JSON compliant
```

Comment (NickM-27) [reply frigate/api/review.py:559]: We are not currently using this in the UI, I think the part commented on should stay the same, perhaps it can be adjusted to work the same as the motion or otherwise we can just delete this API since it is currently not used

Comment (iursevla) [reply frigate/api/review.py:559]: Do you think we can remove it then? @blakeblackshear @hawkeye217 is this okay for you guys too? I prefer to have only code that is used.

File: frigate/api/defs/severity_enum.py
Comment (iursevla) [reply frigate/api/defs/severity_enum.py:1]: @NickM-27 does this mean this code 

```py
 fn.SUM(
                Case(
                    None,
                    [
                        (
                            (ReviewSegment.severity == "significant_motion"),
                            ReviewSegment.has_been_reviewed,
                        )
                    ],
                    0,
                )
            ).alias("reviewed_motion"),
```

Is deprecated or it will be used in the future?

Comment (NickM-27) [reply frigate/api/defs/severity_enum.py:1]: It never existed in production only in development. It might actually be implemented in some future version but it can be removed

Comment (iursevla) [reply frigate/api/defs/severity_enum.py:1]: Can I remove it in BE and FE?

Comment (iursevla) [reply frigate/api/defs/severity_enum.py:1]: @NickM-27 I just did - removed significant_motion usages in the code.

Updated to use the existing enum

File: frigate/api/defs/review_body.py
Comment (NickM-27) [inline frigate/api/defs/review_body.py:4]: Can we combine these to be something generic like ReviewModifyMultipleBody

File: frigate/api/defs/review_query_parameters.py
Comment (NickM-27) [inline frigate/api/defs/review_query_parameters.py:7]: Why was optional removed?

**Review by NickM-27 — CHANGES_REQUESTED** [review]

File: frigate/api/defs/review_query_parameters.py
Comment (iursevla) [reply frigate/api/defs/review_query_parameters.py:7]: Because it would generate wrong documentation with `Optional` (https://github.com/pydantic/pydantic/pull/6653)

This 
```py
cameras: str = "all"
```

It's still optional, with a default value `all`. It will generate the following OpenAPI documentation:

```yaml
 - name: cameras
          in: query
          required: false
          schema:
            type: string
            default: all
            title: Cameras
```

The old code 

```py
cameras: Optional[str] = "all"
```

would generate wrong documentation like:

```yaml
    - name: cameras
          in: query
          required: false
          schema:
            anyOf:
              - type: string
              - type: 'null'
            default: all
            title: Cameras
```

The solutions are:

```py
cameras: str = "all"
```
or
```py
cameras: Union[str, SkipJsonSchema[str]] = "all"
```

For properties that can have the value `None`, then we need the `SkipJsonSchema`

```py
propertyName: Union[<type>, SkipJsonSchema[None]] = None
```

Comment (NickM-27) [reply frigate/api/defs/review_query_parameters.py:7]: gotcha, thanks for the explanation

**Comment by hawkeye217** [comment]
This is looking pretty good. Before approval and merging, can you confirm that each endpoint you've modified works correctly and mimics the behavior of the current code in dev?

**Comment by iursevla** [comment]
> This is looking pretty good. Before approval and merging, can you confirm that each endpoint you've modified works correctly and mimics the behavior of the current code in dev?

@hawkeye217 as far as my tests go, yes. I've done testing using the API docs and everything seems to work correctly.

**Review by NickM-27 — APPROVED** [review]

**Review by hawkeye217 — APPROVED** [review]

**Pull request merged**
